guides/documentations/essentials.md

```diff
@@ -0,0 +1,209 @@
+As you can see, plugin-core can be divided in 5 distinct parts:
+- Metadata managment
+- Generate components from source files (Parser)
+- Generate source files from components (Renderer)
+- Draw diagrams
+- Storage
+
+In each part, we will explain its purpose and its intended function.
+
+## Base of plugin
+
+The base code for a plugin is contained in the `DefaultPlugin` class. This class is the entry point to access all the functionalities of the plugin, like so:
+
+```js
+class DefaultPlugin {
+  constructor(props = {
+    event: null,
+    configuration: null,
+    pluginData: null,
+    pluginDrawer: null,
+    pluginMetadata: null,
+    pluginParser: null,
+    pluginRenderer: null,
+  }) {
+    this.configuration = props.configuration || new DefaultConfiguration();
+    this.data = props.pluginData || new DefaultData(this.configuration, {}, props.event);
+    this.__drawer = props.pluginDrawer || new DefaultDrawer(this.data);
+    this.__metadata = props.pluginMetadata || new DefaultMetadata(this.data);
+    this.__parser = props.pluginParser || new DefaultParser(this.data);
+    this.__renderer = props.pluginRenderer || new DefaultRender(this.data);
+  }
+}
+```
+
+The `DefaultPlugin` provides these methods:
+
+- `init`: Init method, to call once before all plugin usages. Call `this.__metadata.parse()` and `this.data.initLinkDefinitions()`.
+- `initResources`: Set resources in plugin.
+- `resetDrawerActions`: Reset drawer actions.
+- `draw`: Draw all data in the html element defined by the id. Call `this.__drawer.draw()`
+- `parse`: Convert the content of files into plugin data. Configuration file is used for setting up the components' configuration. Call `this.__parser.parse()` and `this.__parser.parseConfiguration()`.
+- `isParsable`: Indicate if this parser can parse this file. Return `this.__parser.isParsable()`.
+- `getModels`: Get models from all project files. Return `this.__parser.getModels(files.filter((file) => this.isParsable(file))`.
+- `render`: Return all generated files from plugin data. Configuration file is used for saving the components' configuration. Call `this.__renderer.renderConfiguration()` and return `this.__renderer.render().concat(configurationFile)`.
+- `arrangeComponentsPosition`: Reorganize nodes layout algorithmically.
+- `repositionComponent`: Reposition a component where there is room for it.
+- `getComponentPosition`: Get the coordinates to use for component draw options.
+- `addComponent`: Create new component with draw option if passed as param. Return `this.data.addComponent()`.
+
+## Metadata management
+
+A component represents a piece of code, related to a specific service, that is defined via the Metadata. It is the building block for creating a diagram. Some components are connected to each other either by links, or because one component may contain another. These actions can be performed through the 'draw' interface or by modifying component attributes in the component panel (in Leto-modelizer).
```

Comment:
Adda sentence to explain that metadata are divide into two parts:
- one with a json files that describe the component as functionnal way
- another technical that plugin core convert previous json into a list of class, that's used into leto-modelizer for list and rules

Reply:
done